docs: add this week in databend 116 (#13425)
datafuselabs/databend: website/blog/2023-10-22-databend-weekly-116.md

For example, the following SQL statement instructs Databend to execute a task called `MyTask1` every day at `6am PST` to insert `(1, 2)` and `(3, 4)` into table `t`:

```sql
CREATE TASK IF NOT EXISTS MyTask1 SCHEDULE = USING CRON '0 6 * * *' 'America/Los_Angeles' COMMENT = 'serverless + cron' AS insert into t (c1, c2) values (1, 2), (3, 4) 
```

The **Background Service** requires Databend Enterprise Edition. Please [contact the Databend team](https://www.databend.com/contact-us) for upgrade information.

If you are interested in learning more, please check out the resources below:

- [PR #13316 | feat: impl create task sql parser and planner](https://github.com/datafuselabs/databend/pull/13316)
- [PR #13344 | feat: add SQL syntax support for alter, execute, describe, show, drop task](https://github.com/datafuselabs/databend/pull/13344)

## Code Corner

Discover some fascinating code snippets or projects that showcase our work or learning journey.

### Understanding Configuration Mapping in Databend

> For a complex database service like Databend, supporting many configurable options helps developers manage and tune the system. A recently published blog explains the mapping between command line options, environment variables, config files, and code in Databend.

Reviewer: and code??

> For a complex database service like Databend, supporting many configurable options helps developers manage and tune the system. A recently published blog explains the mapping between command line options, environment variables, config files, and code in Databend.

Reviewer: ```suggestion
For a complex database service like Databend, supporting many configurable options helps developers manage and tune the system. A recently published blog explains the mappings between command line options, environment variables, config files, and code in Databend.
```

> For a complex database service like Databend, supporting many configurable options helps developers manage and tune the system. A recently published blog explains the mappings between command line options, environment variables, config files, and code in Databend.

Reviewer: ```suggestion
For a complex database service like Databend, supporting many configurable options helps developers manage and tune the system. A recently published blog explains the mappings between command line options, environment variables, and config files by reading the Databend code.
```

Databend currently supports three configuration methods in decreasing order of priority:

> - Command line options are used to override a small number of configs.

Reviewer: 1. Command line options: Override configurations set elsewhere.

> - For K8s clusters, environment variables enable flexibly changing some configs (e.g. feature flags) more elegantly.

Reviewer: 2. Environment Variables: Provide configuration flexibility for Kubernetes clusters.

> - Config files are generally recommended to record and manage configs.

Reviewer: 3. Config files: A recommended approach for recording and managing configurations.

> Notably, Databend supports common environment variables like `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` through the rollback mechanism in [opendal](https://github.com/apache/incubator-opendal) without explicitly declaring them. This simplifies the code and reduces user learning cost.

Reviewer: Please note that Databend also supports common environment variables from storage services such as AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY through the rollback mechanism in [opendal](https://github.com/apache/incubator-opendal). You can directly utilize these variables when working with Databend.

If you are interested in learning more, please check out the resources below:

- [Blog | Navigating Databend's Configuration Maze: A Guide for Developers and Operators](https://databend.rs/blog/2023-10-18-databend-configuration-guide)

## Highlights

We have also made these improvements to Databend that we hope you will find helpful:

> - Added new table function `fuse_encoding` .

Reviewer: ```suggestion
- Added a new table function `fuse_encoding` .
```

- Added new string functions `split` and `split_part`.

> - SQLsmith support for `MERGE INTO`.

Reviewer: ```suggestion
- SQLsmith now supports for `MERGE INTO`.
```

- `databend-metactl` supports viewing cluster status.

## What's Up Next

We're always open to cutting-edge technologies and innovative ideas. You're more than welcome to join the community and bring them to Databend.

### Support More Type Comparisons in z3 Solver

Previously, Databend introduced [z3.rs](https://github.com/prove-rs/z3.rs) to solve constraints by finding solutions that satisfy given conditions. [Z3](https://github.com/Z3Prover/z3) from Microsoft Research is a theorem prover commonly used to solve SMT problems. However, currently Databend's z3 solving only supports integer comparisons and needs to cover more types.

> For example, after supporting string comparison, `t3.f >= '2000-08-23'` in `select * from t1 left join t3 on t3.e = t1.a where t3.f >= '2000-08-23';` can be pushed down to table `t3`.

Reviewer: ```suggestion
For example, after string comparison is supported, `t3.f >= '2000-08-23'` in `select * from t1 left join t3 on t3.e = t1.a where t3.f >= '2000-08-23';` can be pushed down to table `t3`.
```
